modules/process/manager_stacktraces.go

```diff
@@ -0,0 +1,256 @@
+	}
+
+	// We cannot use the process pidmaps here because we have released the mutex ...
+	pidMap := map[IDType]*Process{}
+	processStacks := make([]*Process, 0, len(processes))
+	for _, process := range processes {
+		pStack := &Process{
+			PID:         process.PID,
+			ParentPID:   process.ParentPID,
+			Description: process.Description,
+			Start:       process.Start,
+			Type:        process.Type,
+		}
+
+		pidMap[process.PID] = pStack
+		if flat {
+			processStacks = append(processStacks, pStack)
+		} else if parent, ok := pidMap[process.ParentPID]; ok {
+			parent.Children = append(parent.Children, pStack)
+		} else {
+			processStacks = append(processStacks, pStack)
+		}
+	}
+
+	goroutineCount := int64(0)
+
+	// Now walk through the "Sample" slice in the goroutines stack
+	for _, sample := range stacks.Sample {
+		stack := &Stack{}
+
+		// Add the labels
+		for name, value := range sample.Label {
+			if name == DescriptionPProfLabel || name == PIDPProfLabel || (!flat && name == PPIDPProfLabel) || name == ProcessTypePProfLabel {
+				continue
+			}
+			if len(value) != 1 {
+				// Unexpected...
+				return nil, 0, fmt.Errorf("label: %s in goroutine stack with unexpected number of values: %v", name, value)
+			}
+
+			stack.Labels = append(stack.Labels, &Label{Name: name, Value: value[0]})
+		}
+
+		stack.Count = sample.Value[0]
+		goroutineCount += stack.Count
+
+		// Now get the processStack for this goroutine sample
+		var processStack *Process
+		if pidvalue, ok := sample.Label[PIDPProfLabel]; ok && len(pidvalue) == 1 {
+			pid := IDType(pidvalue[0])
+			processStack, ok = pidMap[pid]
+			if !ok && pid != "" {
+				ppid := IDType("")
+				if value, ok := sample.Label[PPIDPProfLabel]; ok && len(value) == 1 {
+					ppid = IDType(value[0])
+				}
+				description := "(dead process)"
+				if value, ok := sample.Label[DescriptionPProfLabel]; ok && len(value) == 1 {
+					description = value[0] + " " + description
+				}
+				ptype := "code"
+				if value, ok := sample.Label[ProcessTypePProfLabel]; ok && len(value) == 1 {
+					stack.Labels = append(stack.Labels, &Label{Name: ProcessTypePProfLabel, Value: value[0]})
+				}
+				processStack = &Process{
+					PID:         pid,
+					ParentPID:   ppid,
+					Description: description,
+					Type:        ptype,
+				}
+
+				pidMap[processStack.PID] = processStack
+				added := false
+				if processStack.ParentPID != "" && !flat {
+					if parent, ok := pidMap[processStack.ParentPID]; ok {
+						parent.Children = append(parent.Children, processStack)
+						added = true
+					}
+				}
+				if !added {
+					processStacks = append(processStacks, processStack)
+				}
+			}
+		}
+		if processStack == nil {
+			var ok bool
+			processStack, ok = pidMap[""]
+			if !ok {
+				processStack = &Process{
+					Description: "(unassociated)",
+					Type:        "code",
+				}
+				pidMap[processStack.PID] = processStack
+				processStacks = append(processStacks, processStack)
+			}
+		}
```

Comment:
This bit confused me a bit, so this is the "zero PID", only created once in this big function, when the sample has _unexpected_(condition on line 152 fails) data, this is being shown instead?

Reply:
Ah that makes a bit more sense. Then LGTM